src/deps-index.ts

```diff
@@ -21,79 +21,100 @@ export interface DepsIndex {
 }
 
 type DependencyMapEntry = {
-  url: UrlString,
   // Eagerly reachable dependencies.
   eager: Set<UrlString>,
   // Lazily reachable dependencies - may also include some entries from "eager"
-  lazy: Set<UrlString>
+  lazy: Set<UrlString>,
+  // Lazy edges are those imports explicitly lazy-imported
+  lazyEdges: Set<UrlString>,
 };
 
-async function _getTransitiveDependencies(
-    url: UrlString, entrypoints: UrlString[], analyzer: Analyzer):
-    Promise<DependencyMapEntry> {
-      const analysis = await analyzer.analyze([url]);
-      const document = analysis.getDocument(url);
-      if (!(document instanceof Document)) {
-        const message = document && document.message || 'unknown';
-        throw new Error(`Unable to get document ${url}: ${message}`);
-      }
-      const imports = document.getFeatures(
-          {kind: 'import', externalPackages: true, imported: true});
-      const eagerImports = new Set<UrlString>();
-      const lazyImports = new Set<UrlString>();
-      for (const htmlImport of imports) {
-        try {
-          console.assert(
-              htmlImport.url, 'htmlImport: %s has no url', htmlImport);
-        } catch (err) {
-          if (err instanceof AssertionError) {
-            continue;
-          }
-          throw err;
-        }
+/**
+ * For a given URL, return a set of eager and lazy dependencies.  The
+ * function is recursive, but a `visited` set is carried forward to short-
+ * circuit loops encountered.
+ */
+function getTransitiveDependencies(
+    url: UrlString,
+    analysis: Analysis,
+    visited: Set<UrlString> = new Set<UrlString>()): DependencyMapEntry {
+  const map: DependencyMapEntry = {
+    eager: new Set(),
+    lazy: new Set(),
+    lazyEdges: new Set()
+  };
+  const document = analysis.getDocument(url);
+  if (!(document instanceof Document)) {
```

Comment:
I *kinda* want an `analysis.getDocumentOrBust(url);` option to dry up these redundant go-like error-handling clauses.